src/lib.rs

```diff
@@ -19,14 +42,30 @@ pub trait InputSwitch {
     /// use switch_hal::{InputSwitch, OutputSwitch, Switch, IntoSwitch};
     /// # let pin = mock::Pin::with_state(mock::State::High);
     /// # let mut status_led = mock::Pin::new().into_active_high_switch();
-    /// let button = pin.into_active_low_switch();
+    /// let mut button = pin.into_active_low_switch();
```

Comment:
This example highlights why I question the mutability. It makes no sense for a button input to be mutable. For the application code, after setup is complete, it’s a read only peripheral.

Reply:
Unless you have a Debouncer pin wrapper (maintains state) for example. Hidden mutability is then problematic for multi threading etc. Same for port expanders (again, hidden mutability is needed for the underlying bus).

I would have been OK with both approaches, but this one is more explicitly saying that sharing is not guaranteed.

However that ship sailed 10 months ago, the API is stabilized and I am more affected by switch-hal still depending 0.2 than by the mut :) I can work around the dependency, but it is annoying.

Reply:
All the rest of this PR has already been changed to use interior mutability; this mut can be removed now.